### PR TITLE
Backport NgOptimizedImage to v13

### PR DESCRIPTION
**Pull request by atcastle**
This PR is an extension of the work done in #49278. This extend the backport of NgOptimizedImage from version 14 to version 13.

To aid in review, here's a list of additional changes made, beyond those made in the version 14 backport:

`ng_optimized_image.ts`: 
* Change @usageNotes to remove references to standalone
* Remove conditional `host` property from @Directive, because this does not behave the same pre-14.
* Replaced the above functionality with calls to `this.renderer.setStyle`
* Move injected values for class properties from the class body to within the constructor, using @Inject.
* Export NgModule at end of file

`preconnect_link_checker.ts`:
* Move some injected properties from class body to constructor
* Directly access InjectFlags enum, as handling of `optional` param in `inject` config is different pre-14

`preload-link-creator.ts`:
* Move injected properties from class body to constructor

All built-in loader files (Imgix, Cloudflare, etc):
* Export create[Service]Url function to facilitate testing

**Testing**
`render3.ts`:
* Backport utility functions `withHead`, and `wrapTest`

`ng_optimized_image_spec.ts`:
* Use NgOptimizedImageModule instead of standalone

`image_loader_spec.ts`:
* These tests rely on EnvironmentInjector to access the image loaders for testing. I replaced them with somewhat-hacky logic for stubbing out the loader functions, using the newly-exported functions in the loader files.

e2e tests:
* All test apps modified to use NgOptimizedImageModule instead of standalone

**Docs**
References to using NgOptimizedImage in standalone removed.

Note that there may be e2e failures in this PR initially, as I'm unable to run the pre-Angular15 e2e tests locally.

CC: @AndrewKushnir @pkozlowski-opensource @kara

### CODE REVIEW

**Comment by devversion** [comment]
There seem to be legitimate failures. Please ping if this is ready for review. Converting to draft in the meanwhile.

File: aio/content/guide/image-directive.md [hunk 1]
```diff
@@ -0,0 +1,323 @@
+<div class="alert is-important">
+Notice: Angular 14.3.0 includes a backported version of the NgOptimizeImage directive present in later versions of Angular. This was implemented by the Chrome Aurora team, as a special project to make performance enhancements available to more applications built on Angular. This does not reflect a shift in Angular release procedure and does not indicate that any additional features will be backported to earlier Angular versions.
```
Comment (AndrewKushnir) [inline aio/content/guide/image-directive.md:2]:
```suggestion
Notice: Angular 13.3.0 includes a backported version of the NgOptimizeImage directive present in later versions of Angular. This was implemented by the Chrome Aurora team, as a special project to make performance enhancements available to more applications built on Angular. This does not reflect a shift in Angular release procedure and does not indicate that any additional features will be backported to earlier Angular versions.
```

File: aio/content/guide/image-directive.md [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+*   Warning if `width` or `height` have been set incorrectly
+*   Warning if the image will be visually distorted when rendered
+
+## Getting Started
+
+#### Step 1: Import NgOptimizedImage
+
+<code-example format="typescript" language="typescript">
+
+import { NgOptimizedImageModule } from '@angular/common'
```
Comment (AndrewKushnir) [inline aio/content/guide/image-directive.md:30]:
```suggestion
#### Step 1: Import NgOptimizedImageModule

<code-example format="typescript" language="typescript">

import { NgOptimizedImageModule } from '@angular/common';
```

File: aio/content/navigation.json [hunk 3]
```diff
@@ -786,6 +786,17 @@
         }
       ]
     },
+    {
```
Comment (AndrewKushnir) [inline aio/content/navigation.json:789]: I think we should include the link in the navigation in a similar section like we have on angular.io (this PR adds a "Feature preview" section, which is likely not needed).

File: packages/common/src/directives/ng_optimized_image/ng_optimized_image.ts [hunk 4]
```diff
@@ -0,0 +1,1012 @@
...
+  private lcpObserver;
+  private imgElement: HTMLImageElement;
+
+  constructor(
+      @Inject(IMAGE_LOADER) private imageLoader: ImageLoader,
+      @Inject(IMAGE_CONFIG) private config: ImageConfig,
+      @Inject(Renderer2) private renderer: Renderer2,
+      @Inject(ElementRef) private elementRef: ElementRef,
+      @Inject(Injector) private injector: Injector,
+      @Inject(PLATFORM_ID) private isServer: any,
```
Comment (AndrewKushnir) [inline packages/common/src/directives/ng_optimized_image/ng_optimized_image.ts:349]:
```suggestion
      @Inject(PLATFORM_ID) private platformId: string,
```

**Review by AndrewKushnir — COMMENTED** [review]
@atcastle looks great, just left a few minor comments.

File: aio/content/guide/image-directive.md (hunk 1 above)
Comment (atcastle) [reply aio/content/guide/image-directive.md]: Done

File: aio/content/guide/image-directive.md (hunk 2 above)
Comment (atcastle) [reply aio/content/guide/image-directive.md]: Done

File: packages/common/src/directives/ng_optimized_image/ng_optimized_image.ts (hunk 4 above)
Comment (atcastle) [reply packages/common/src/directives/ng_optimized_image/ng_optimized_image.ts]: Done

**Review by AndrewKushnir — DISMISSED** [review]
Looks great! 👍

File: aio/content/guide/image-directive.md [hunk 5]
```diff
@@ -0,0 +1,323 @@
+<div class="alert is-important">
+Notice: Angular 13.4.0 includes a backported version of the NgOptimizeImage directive present in later versions of Angular. This was implemented by the Chrome Aurora team, as a special project to make performance enhancements available to more applications built on Angular. This does not reflect a shift in Angular release procedure and does not indicate that any additional features will be backported to earlier Angular versions.
```
Comment (kara) [inline aio/content/guide/image-directive.md:2]:
```suggestion
Notice: Angular 13.4.0 includes a backported version of the NgOptimizedImage directive present in later versions of Angular. This was implemented by the Chrome Aurora team, as a special project to make performance enhancements available to more applications built on Angular. This does not reflect a shift in Angular release procedure and does not indicate that any additional features will be backported to earlier Angular versions.
```

**Review by kara — DISMISSED** [review]
LGTM, just one typo

File: aio/content/guide/image-directive.md (hunk 5 above)
Comment (atcastle) [reply aio/content/guide/image-directive.md]: good catch

**Review by kara — APPROVED** [review]

**Review by AndrewKushnir — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
@devversion this PR is ready for merge, but we didn't add the "merge" label, so that it didn't show up in the Caretaker queue.

The Pullapprove also wants an LGTM from you, could you please have a quick look at the DevInfra-related changes? I think we can skip DevRel's review as we are copy'n'pasting docs from the previous backport and an overall plan was LGTM'ed previously by DevRel.

**Review by josephperrott — APPROVED** [review]
LGTM

**Review by josephperrott — APPROVED** [review]
LGTM

Reviewed-for: global-approvers

**Pull request merged**